Documentation for dynamic scaling of Argo CD application controller. (#67913)
openshift/openshift-docs: modules/gitops-argo-cd-dynamic-scaling-on-web-console.adoc

// Module included in the following assemblies:
//
// * declarative_clusterconfig/sharding-clusters-across-argo-cd-application-controller-replicas.adoc

:_mod-docs-content-type: PROCEDURE

> [id="gitops-argo-cd-dynamic-scaling-on-web-console_{context}"]
> = Enabling dynamic scaling of shards on the web console

Reviewer: ```suggestion
[id="gitops-argo-cd-dynamic-scaling-in-web-console_{context}"]
= Enabling dynamic scaling of shards in the web console
```

Author: ACK

> You can enable dynamic scaling of shards using the {OCP} web console.

Reviewer: ```suggestion
You can enable dynamic scaling of shards by using the {OCP} web console.
```

Author: ACK

.Prerequisites
* You have access to the cluster with `cluster-admin` privileges.
* You have access to the {OCP} web console.
* You have installed the {gitops-title} Operator in your cluster.

.Procedure

> . In the *Administator* perspective of the {OCP} web console, go to *Operators* → *Installed Operators*.

Reviewer: ```suggestion
. In the *Administator* perspective of the {OCP} web console, go to *Operators* -> *Installed Operators*.
```

. From the the list of *Installed Operators*, select the {gitops-title} Operator, and then click on the *ArgoCD* tab.

> . Select the Argo CD instance name for which you want to enable dynamic scaling of shards, for example `openshift-gitops`.

Reviewer: ```suggestion
. Select the Argo CD instance name for which you want to enable dynamic scaling of shards, for example, `openshift-gitops`.
```

Author: ACK

. Click on the *YAML* tab, and then edit and configure the `spec.controller.sharding` properties as follows:
+
.Example Argo CD YAML file with dynamic scaling enabled
[source,yaml]
----
apiVersion: argoproj.io/v1beta1
kind: ArgoCD
metadata:
  name: openshift-gitops
  namespace: openshift-gitops
spec:
  controller:
    sharding:
      dynamicScalingEnabled: true <1>
      minShards: 1 <2>
      maxShards: 3 <3>
      clustersPerShard: 1 <4>
----
<1> Set `dynamicScalingEnabled` to `true` to enable dynamic scaling.
<2> Set `minShards` to the minimum number of shards that you want to have. The value must be set to `1` or greater.
<3> Set `maxShards` to the maximum number of shards that you want to have. The value must be greater than the value of `minShards`.
<4> Set `clustersPerShard` to the number of clusters that you want to have per shard. The value must be set to `1` or greater.

. Click *Save*.
+

> You get a success notification alert that says: `<instance_name> has been updated to version <version>`.

Reviewer: ```suggestion
A success notification alert, `openshift-gitops has been updated to version <version>`, appears.
```

Author: ACK

+
[NOTE]
====

> If you edit the default openshift-gitops instance, the *Managed resource* dialog box is displayed. Click *Save* again to confirm the changes.

Reviewer: ```suggestion
If you edit the default `openshift-gitops` instance, the *Managed resource* dialog box is displayed. Click *Save* again to confirm the changes.
```

Author: ACK

====

> . Verify that sharding is enabled by checking the number of pods in the namespace.

Reviewer: ```suggestion
. Verify that sharding is enabled by checking the number of pods in the namespace:
```

Author: ACK

.. Go to *Workloads* -> *StatefulSets*.

> .. Select the namespace where the Argo CD instance is deployed from the *Project* drop-down list, for example `openshift-gitops`.

Reviewer: ```suggestion
.. Select the namespace where the Argo CD instance is deployed from the *Project* drop-down list, for example, `openshift-gitops`.
```

Author: ACK

> .. Click on the name of the StatefulSet that has the name of the Argo CD instance, for example `openshift-gitops-apllication-controller`.

Reviewer: ```suggestion
.. Click on the name of the `StatefulSet` object that has the name of the Argo CD instance, for example `openshift-gitops-apllication-controller`.
```

Author: ACK
